File: ToDoList/app/src/main/java/com/davdo/todolist/src/NoteTypeConverters.java
package com.davdo.todolist.src;

import androidx.room.TypeConverter;

import java.util.Date;
import java.util.UUID;

public class NoteTypeConverters {


    @TypeConverter
    public long fromDate(Date d) {

        if(d != null)
            return d.getTime();

        return -1;
    }

    @TypeConverter
    public Date toDate(long t) {

        return new Date(t);
    }

    @TypeConverter
    public String fromUUID(UUID id) {

        if(id != null)
            return id.toString();

        return null;
    }

    @TypeConverter
    public UUID toUUID(String id) {

        return UUID.fromString(id);
    }

}

File: ToDoListKotlin/app/src/main/java/com/davdo/todolistkotlin/db/NoteTypeConverters.kt
package com.davdo.todolistkotlin.db

import androidx.room.TypeConverter
import java.util.*

class NoteTypeConverters {

	@TypeConverter
	fun fromDate(date: Date?): Long? {
		return date?.time
	}

	@TypeConverter
	fun toDate(ms: Long?) : Date? {

		return ms?.let { Date(it) }
	}

	@TypeConverter
	fun toUUID(uuid: String?): UUID? {
		return UUID.fromString(uuid)
	}

	@TypeConverter
	fun fromUUID(uuid: UUID?) : String? {
		return uuid?.toString()
	}
}